src/couch/src/couch_httpd_auth.erl

```diff
@@ -189,7 +189,7 @@ proxy_auth_user(Req) ->
             Roles =
                 case header_value(Req, XHeaderRoles) of
                     undefined -> [];
-                    Else -> [?l2b(R) || R <- string:tokens(Else, ",")]
+                    Else -> re:split(Else, "\\s*,\\s*", [trim, {return, binary}])
```

Comment:
Did we already strip whitespace on both sides of the input (`Else`)?  Otherwise we get different behavior there:

> re:split(" a , b,c ,d ", "\\s*,\\s*", [trim, {return, list}]).
[" a","b","c","d "]

> string:tokens(" a  , b,c, d ", " ,").
["a","b","c","d"]

Reply:
Yup, that seems to work

 ```
> string:trim("\n\tA \n\r\tB\n\n\t\r\n \t").
"A \n\r\tB"
```

Reply:
Let's skip as discussed in `#dev` slack since we didn't have that behavior previously.

Eventually we could try to make a PR for mochiweb to update header parsing there or have our own wrapper that trims the whitespace for header values in something like couch_httpd:get_header